src/preset.ts

```diff
@@ -836,12 +836,24 @@ export async function expandConfigurePreset(folder: string, name: string, worksp
     // [Windows Only] If CMAKE_CXX_COMPILER or CMAKE_C_COMPILER is set as cl, clang, clang-cl, clang-cpp and clang++,
     // but they are not on PATH, then set the env automatically.
     if (process.platform === 'win32') {
-        if (preset.cacheVariables) {
-            const cxxCompiler = getStringValueFromCacheVar(preset.cacheVariables['CMAKE_CXX_COMPILER'])?.toLowerCase();
-            const cCompiler = getStringValueFromCacheVar(preset.cacheVariables['CMAKE_C_COMPILER'])?.toLowerCase();
+        if (preset.cacheVariables && expandedPreset.cacheVariables) {
+            const cxxCompilerPath = getStringValueFromCacheVar(expandedPreset.cacheVariables['CMAKE_CXX_COMPILER']);
+            const cCompilerPath = getStringValueFromCacheVar(expandedPreset.cacheVariables['CMAKE_C_COMPILER']);
+            const cxxCompiler = path.basename(cxxCompilerPath || "").toLocaleLowerCase();
+            const cCompiler = path.basename(cCompilerPath || "").toLocaleLowerCase();
             // The env variables for the supported compilers are the same.
             const compilerName: string | undefined = util.isSupportedCompiler(cxxCompiler) || util.isSupportedCompiler(cCompiler);
 
+            // If CMake receives an absolute path to cl.exe it tries to perform a compiler check which is undesirable
```

Comment:
Using the CMAKE_<LANG>_COMPILER_WORKS is risky. It is intentionally not documented by CMake. We don't want our implementation to rely on undocumented variables. 

What is the case that you're trying to solve? I've tested with a preset that uses an absolute path to the compiler and it seems to configure and build successfully.

Reply:
I've made an observation, that CMake check fails on otherwise valid compiler without a good reason. I have an assumption that it might be related to CMake version. Anyway, what I recommend to do is to @qarni to incorporate my changes, since there is only a husk of my original intentions left, and to avoid merging, and to close this issue.

Reply:
@oreyg Sorry for the delay, but just following up. We will make sure that when we get back to implementing @qarni's PR, we will keep this PR in mind and make sure that the changes this PR has that we need are incorporated. 
THanks!